src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2101,19 +2099,21 @@ private bool RefMustBeObeyed(bool isDelegateCreation, ArgumentSyntax argumentSyn
 
         private BoundExpression BindArgumentValue(DiagnosticBag diagnostics, ArgumentSyntax argumentSyntax, bool allowArglist, RefKind refKind)
         {
-            if (argumentSyntax.Expression != null)
+            var declarationExpression = argumentSyntax.Expression as DeclarationExpressionSyntax;
```

Comment:
We were just discussion type checks. In this case, it seems that a kind check is possible with `SyntaxKind.DeclarationExpression`. Would that be preferable?